src/FakeItEasy/Core/FakeManager.CancellationRule.cs

```diff
@@ -0,0 +1,51 @@
+namespace FakeItEasy.Core
+{
+    using System;
+    using System.Linq;
+#if FEATURE_NETCORE_REFLECTION
+    using System.Reflection;
+#endif
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <content>Event rule.</content>
+    public partial class FakeManager
+    {
+#if FEATURE_BINARY_SERIALIZATION
+        [Serializable]
+#endif
+        private class CancellationRule : IFakeObjectCallRule
+        {
+            public int? NumberOfTimesToCall => null;
+
+            public bool IsApplicableTo(IFakeObjectCall fakeObjectCall) =>
+                fakeObjectCall.Arguments.OfType<CancellationToken>().Any(ct => ct.IsCancellationRequested);

```

Comment:
Oops, I just realized something... This will match any argument of type `CancellationToken`, *even if the parameter isn't declared as such* (e.g. if it's declared as `object`). I'll update it to take the actual parameter type into account.